daceml/onnx/op_implementations/img_op_implementations.py

```diff
@@ -0,0 +1,360 @@
+
+        if "Indices" in {e.src_conn for e in state.out_edges(node)}:
+            return False
+
+        image_dims = len(X.shape) - 2
+
+        # only do 2D for now
+        if image_dims != 2:
+            return False
+
+        if node.pads is not None and (not all(p == 0 for p in node.pads)
+                                      or len(node.pads) != image_dims * 2):
+            return False
+
+        if node.strides is not None and len(node.strides) != image_dims:
+            return False
+
+        if node.auto_pad != 'NOTSET':
+            return False
+
+        if node.ceil_mode != 0 or node.storage_order != 0:
+            return False
+
+        if node.dilations is not None and (not all(d == 1
+                                                   for d in node.dilations) or
+                                           len(node.dilations) != image_dims):
+            return False
+        return True
+
+    @staticmethod
+    def forward(node: ONNXOp, state: SDFGState,
+                sdfg: SDFG) -> typing.Union[nodes.Node, SDFG]:
+        X = in_desc_with_name(node, state, sdfg, "X")
+        Y = out_desc_with_name(node, state, sdfg, "Y")
+
+        image_dims = len(X.shape) - 2
+        batch_size = X.shape[0]
+        num_channels = X.shape[1]
+        strides = node.strides if node.strides is not None else [
+            1 for _ in range(image_dims)
+        ]
+        stride_x, stride_y = strides
+        filter_hx, filter_hy = node.kernel_shape
+        output_size_y, output_size_x = Y.shape[2:]
+
+        new_sdfg = dace.SDFG("pure_maxpool")
+
+        init_state = new_sdfg.add_state("init")
+
+        new_state = new_sdfg.add_state_after(init_state, "compute")
+        new_sdfg.add_datadesc("X", copy.deepcopy(X))
+        new_sdfg.add_datadesc("Y", copy.deepcopy(Y))
+
+        new_sdfg.arrays["X"].transient = False
+        new_sdfg.arrays["Y"].transient = False
+
+        # add init state
+        # yapf: disable
+        init_state.add_mapped_tasklet("init",
+                                      map_ranges={
+                                          "i{}".format(i): "0:{}".format(s)
+                                          for i, s in enumerate(Y.shape)
+                                      },
+                                      inputs={},
+                                      code="y = {}".format(dtypes.min_value(Y.dtype)),
+                                      outputs=dict(
+                                          y=dace.Memlet("Y[{}]".format(
+                                              ", ".join("i{}".format(i)
+                                                        for i, _ in enumerate(Y.shape))))
+                                      ),
+                                      external_edges=True)
+        # yapf: enable
+
+        # the outer map loops over every entry in the output array
+        outer_me, outer_mx = new_state.add_map(
+            'outer_conv_map',
```

Comment:
not exactly conv (also below)

Reply:
@orausch don't you want to change it to `outer_pool_map` etc.?

Reply:
oops, changed